utils/integration-test-utils/src/integration_test_utils.rs

```diff
@@ -58,41 +55,46 @@ pub async fn check_send_receive_message_discrepancy(
     }
 }
 
+// TODO: Fix this function to work with UUIDv7
 pub async fn check_messages_in_order(messages: Arc<Mutex<Vec<UMessage>>>) {
     let messages = messages.lock().await;
     if messages.is_empty() {
         return;
     }
 
-    // Step 1: Group messages by id.lsb
-    let mut grouped_messages: HashMap<u64, Vec<&UMessage>> = HashMap::new();
+    // Step 1: Group messages by source UUri
+    #[allow(clippy::mutable_key_type)]
+    let mut grouped_messages: HashMap<UUri, Vec<&UMessage>> = HashMap::new();
     for msg in messages.iter() {
-        let lsb = msg.attributes.as_ref().unwrap().id.lsb;
-        grouped_messages.entry(lsb).or_default().push(msg);
+        let source_uuri = msg
+            .attributes
+            .as_ref()
+            .unwrap()
+            .source
+            .as_ref()
+            .unwrap()
+            .clone();
+        grouped_messages.entry(source_uuri).or_default().push(msg);
     }
 
-    // Step 2: Check each group for strict increasing order of id.msb
-    for (lsb, group) in grouped_messages {
-        debug!("lsb: {lsb}");
+    // Step 2: Check each group for strict increasing order of id.msb first 48 bytes
+    for (source_uuri, group) in grouped_messages {
+        debug!("source_uuri: {source_uuri}");
         if let Some(first_msg) = group.first() {
-            let mut prev_msb = first_msg.attributes.as_ref().unwrap().id.msb;
+            let mut prev_timestamp = first_msg.attributes.as_ref().unwrap().id.msb >> 16;
```

Comment:
maybe https://docs.rs/up-rust/0.1.5/up_rust/struct.UUID.html#method.get_time comes in handy here?

Reply:
Good point, I should use this. Will do in follow-up